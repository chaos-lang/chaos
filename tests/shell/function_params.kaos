// Wrong parameter type
void def list_as_params(str param1, list param2)
    print param1
    print param2
end

list_as_params('foo', 'bar')

void def dict_as_params(str param1, dict param2)
    print param1
    print param2
end

dict_as_params('foo', 'bar')


// Illegal element type for a typed list or dictionary param
void def typed_list_as_params(str param1, num list param2)
    print param1
    print param2
    param2[0] = 'a'
    print param2
end

typed_list_as_params('foo', ['a', 2, 3])
typed_list_as_params('foo', [1, 2, 3])

void def typed_dict_as_params(str param1, num dict param2)
    print param1
    print param2
    param2['a'] = 'a'
    print param2
end

typed_dict_as_params('foo', {'a': 'a', 'b': 2, 'c': 3})
typed_dict_as_params('foo', {'a': 1, 'b': 2, 'c': 3})


// Wrong typed list or dictionary
void def typed_list_as_params(str param1, num list param2)
    print param1
    print param2
end

typed_list_as_params('foo', ['a', 2, 3])
typed_list_as_params('foo', [1, 2, 3])

void def typed_dict_as_params(str param1, num dict param2)
    print param1
    print param2
end

typed_dict_as_params('foo', {'a': 'a', 'b': 2, 'c': 3})
typed_dict_as_params('foo', {'a': 1, 'b': 2, 'c': 3})
