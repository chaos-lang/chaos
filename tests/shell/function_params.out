Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 7
list_as_params('foo', 'bar')
Illegal variable type for function parameter: param2 of function: list_as_params
Absorbed by Interactive Shell
Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 14
dict_as_params('foo', 'bar')
Illegal variable type for function parameter: param2 of function: dict_as_params
Absorbed by Interactive Shell
Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 21
param2[0] = 'a'
Illegal element type: String for the typed list: param2
Absorbed by Interactive Shell
Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 25
typed_list_as_params('foo', ['a', 2, 3])
Undefined function: typed_list_as_params in <module>
Absorbed by Interactive Shell
Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 26
typed_list_as_params('foo', [1, 2, 3])
Undefined function: typed_list_as_params in <module>
Absorbed by Interactive Shell
Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 31
param2['a'] = 'a'
Illegal element type: String for the typed list: param2
Absorbed by Interactive Shell
Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 35
typed_dict_as_params('foo', {'a': 'a', 'b': 2, 'c': 3})
Undefined function: typed_dict_as_params in <module>
Absorbed by Interactive Shell
Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 36
typed_dict_as_params('foo', {'a': 1, 'b': 2, 'c': 3})
Undefined function: typed_dict_as_params in <module>
Absorbed by Interactive Shell
Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 45
typed_list_as_params('foo', ['a', 2, 3])
Illegal variable type for function parameter: param2 of function: typed_list_as_params
Absorbed by Interactive Shell
foo
[1, 2, 3]
Chaos Error (most recent call last):
File: "~/chaos/__interactive__.kaos", line 53
typed_dict_as_params('foo', {'a': 'a', 'b': 2, 'c': 3})
Illegal variable type for function parameter: param2 of function: typed_dict_as_params
Absorbed by Interactive Shell
foo
{'a': 1, 'b': 2, 'c': 3}
